src/Sarif.WorkItems/SarifWorkItemsExtensions.cs

```diff
@@ -83,22 +83,54 @@ public static string CreateWorkItemTitle(this Run run)
                     (locationName == null ? "" : " (in " + locationName + ")");
         }
 
-        public static Dictionary<Run, int> ComputeRunResultCounts(this SarifLog log)
+        public static List<string> GetRunToolNames(this SarifLog log)
         {
-            if (log == null) { throw new ArgumentNullException(nameof(log)); }
-            if (log.Runs == null) { throw new ArgumentNullException(nameof(log.Runs)); }
+            return log?.Runs?.Where(r => r?.Results?.Count > 0).Select(r => r.Tool?.Driver?.Name).Distinct().ToList();
+        }
 
-            var resultCountsByRun = new Dictionary<Run, int>();
+        public static int GetAggregateResultCount(this SarifLog log)
+        {
+            return log?.Runs?.Select(r => r?.Results?.Count).Sum() ?? 0;
+        }
 
-            foreach (Run run in log?.Runs)
+        public static string PhraseToolNames(List<string> toolNames)
+        {
+            toolNames = toolNames.Where(r => !string.IsNullOrWhiteSpace(r)).ToList();
+            if (toolNames.Count == 0)
+            {
+                return string.Empty;
+            }
+            else if (toolNames.Count == 1)
+            {
+                return string.Format("'{0}'", toolNames[0]);
+            }
+            else
             {
-                if (run?.Results != null)
+                string phrasedToolNames = string.Format("'{0}'", toolNames[0]);
+                for (int j = 1; j < toolNames.Count - 1; j++)
                 {
-                    resultCountsByRun.Add(run, run.Results.Count);
+                    phrasedToolNames = string.Join(", ", phrasedToolNames, string.Format("'{0}'", toolNames[j]));
                 }
+                phrasedToolNames = string.Join(" and ", phrasedToolNames, string.Format("'{0}'", toolNames[toolNames.Count - 1]));
+                return phrasedToolNames;
             }
+        }
+
+        public static string CreateWorkItemDescription(this SarifLog log, Uri locationUri)
+        {
+            List<string> toolNames = log.GetRunToolNames();
+            int totalResults = log.GetAggregateResultCount();
+            string phrasedToolNames = PhraseToolNames(toolNames);
+            string multipleToolsFooter = toolNames.Count > 1 ? WorkItemsResources.MultipleToolsFooter : string.Empty;
+
+            Uri runRepositoryUri = log?.Runs.FirstOrDefault()?.VersionControlProvenance?.FirstOrDefault().RepositoryUri;
+            string detectionLocation = !string.IsNullOrEmpty(runRepositoryUri?.OriginalString) ? runRepositoryUri?.OriginalString : locationUri?.OriginalString;
+          
+            var descriptionText = new StringBuilder(string.Format(WorkItemsResources.WorkItemBodyTemplateText, totalResults, phrasedToolNames, detectionLocation, multipleToolsFooter));
```

Comment:
>descriptionText [](start = 16, length = 15)

Naming nit: I prefer "builder" in the name when using a string builder. otherwise it looks like you're trying to append to a string variable. So: `descriptionBuilder`.